### PR TITLE
Return value from realm.write

### LINKED ISSUE #2237: Return value from realm.write
Allow returning value from `realm.write` 

E.g.

```js
const user = realm.write(() => {
  return realm.create('User', {})
})
```

If you accept this feature request, I can provide PR.

### PR DESCRIPTION
**Pull request by zepedr0**
## What, How & Why?

Allow returning value from realm.write

E.g.

```js
const user = realm.write(() => {
  return realm.create('User', {})
})
```

This closes #2237 

## ☑️ ToDos

* [x] 📝 Changelog entry
* ~[ ] 📝 `Compatibility` label is updated or copied from previous entry~
* [x] 🚦 Tests
* [x] 📝 Public documentation PR created or is not necessary
* ~[ ] 💥 `Breaking` label has been applied or is not necessary~

*If this PR adds or changes public API's:*
* [x] typescript definitions file is updated
* [x] jsdoc files updated
* ~[ ] Chrome debug API is updated if API is available on React Native~

### CODE REVIEW

File: types/index.d.ts [hunk 1]
```diff
@@ -779,10 +779,10 @@ declare class Realm {
     removeAllListeners(name?: string): void;
 
     /**
-     * @param  {()=>void} callback
-     * @returns void
+     * @param  {()=>ReturnValueType} callback
+     * @returns {ReturnValueType}
      */
-    write(callback: () => void): void;
+    write<ReturnValueType>(callback: () => ReturnValueType): ReturnValueType;
```
Comment (kraenhansen) [inline types/index.d.ts:785]: Does this `ReturnValueType` need a default value or is it able to infer this entirely from the callback? I would have thought you needed to use the `infer` keyword or `ReturnType` helper type or at least `write<ReturnValueType = unknown>(...)`.

Comment (takameyer) [reply types/index.d.ts:785]: I worked on this with José.  The type of the return value is inferred from the callback return value (we tested this).  That being said, the default value should probably be set to unknown (the value is just typed as `void` if nothing is returned).

**Review by takameyer — APPROVED** [review]
👍🏼

**Comment by takameyer** [comment]
As in the other PR, the changelog should be updated.

File: types/index.d.ts (hunk 1 above)
Comment (kraenhansen) [reply types/index.d.ts:785]: I just tested it in the TS playground - it works as I expect it to. Good job!
https://www.typescriptlang.org/play?#code/CYUwxgNghgTiAEAzArgOzAFwJYHtXwHcYsMQAeAJQD4AKMKCCAIyjAGsAueGgSngF4q8Cjy4UA3AChJGAJ4AHBAFUAziBgD4Ab0nw98VFAC2ILiozFUAcykBfaWDzn4yNRv6FipGrwFCd+vCOqM6u6lyq6poBgfqGJlwARABSOAAWqIkANLr6tlKBcBjIMPhhMHY8UpLBKjgQIAB0EDhWNOWN8SBVkkA

**Review by kneth — APPROVED** [review]

**Review by fronck — APPROVED** [review]

**Review by kraenhansen — APPROVED** [review]
Much better :)

**Review by takameyer — APPROVED** [review]

**Pull request merged**